src/main/java/bio/terra/service/filedata/flight/ingest/IngestPopulateFileStateFromFileStep.java

```diff
@@ -0,0 +1,102 @@
+package bio.terra.service.filedata.flight.ingest;
+
+import bio.terra.model.BulkLoadFileModel;
+import bio.terra.model.BulkLoadRequestModel;
+import bio.terra.service.filedata.exception.BulkLoadControlFileException;
+import bio.terra.service.filedata.google.gcs.GcsBufferedReader;
+import bio.terra.service.job.JobMapKeys;
+import bio.terra.service.load.LoadService;
+import bio.terra.service.load.flight.LoadMapKeys;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+// Populate the files to be loaded from the incoming array
+public class IngestPopulateFileStateFromFileStep implements Step {
+    // TODO: MAYBE make these configuration items?
```

Comment:
They should at least be config items. `MAX_BAD_LINES` seems like it could be useful to expose in the request.

Reply:
I can make them config items. The reason to have MAX_BAD_LINES is so we don't build an enormous errorDetail array in the ErrorModel. I don't think we want to have it settable by the caller. If there are syntax errors in the file, we don't load any data, so this isn't like the maxFailedLoads.